src/eth_provider/provider.rs

```diff
@@ -153,20 +134,18 @@ where
         let sort = doc! { "header.number": -1 };
         let header: Option<StoredHeader> = self.database.get_one("headers", filter, sort).await?;
         let block_number = match header {
-            None => U64::from(self.starknet_provider.block_number().await?), // in case the database is empty, use the starknet provider
+            None => U64::from(self.starknet_provider.block_number().await.map_err(KakarotError::from)?), // in case the database is empty, use the starknet provider
             Some(header) => {
-                let number = header.header.number.ok_or(EthProviderError::ValueNotFound("Block".to_string()))?;
-                let number: u64 = number
-                    .try_into()
-                    .map_err(|_| ConversionError::ValueOutOfRange("Block number too large".to_string()))?;
+                let number = header.header.number.ok_or(EthApiError::UnknownBlockNumber)?;
+                let number: u64 = log_map_err(number.try_into(), |_| EthApiError::UnknownBlockNumber)?;
```

Comment:
i'm not a fan of this API / naming

it's not so intuitive that the first argument is "tried" and the second is exception / the mapped error

Reply:
if you have a better API lmk, I think keeping as close as possible to the result API makes sense

Reply:
I guess I understand your point, isn't there a way to add logging natively to errors? Kinda like a hook or smth?
Seems a bit backwards

Reply:
Since rust `1.76`, `inspect_err` has been stabilized and looks like what we would want.